### ISSUE TITLE
Steam Controller doesn't work with video driver not "vulkan"

### ISSUE THREAD

**Issue by ml-** [libretro/RetroArch#9112]
## Description

When using Steam Big Picture Mode with Steam Controller and RetroArch added as a Shortcut, the proper Controller settings are only if video driver is "vulkan" and input driver is "udev".

Changing input driver makes the Input work, but not as a gamepad. It takes the Desktop configuration. Left touchpad to scroll, right touchpad mouse.

### Expected behavior

Steam Controller behaves with other cores the same as with "vulkan".

### Actual behavior

No input when using Steam Controller. Other Controllers configures through Steam Big Picture Mode work fine. Tested with a PS2-like USB Controller.

### Steps to reproduce the bug

1. Add RetroArch as a non-Steam game shortcut to the library
2. Connect Steam Controller
3. Big Picture Mode
4. Run RetroArch
5. Change Video driver to "gl"
6. Restart RetroArch
7. Input with Steam Controller no longer works.

### Version/Commit

- RetroArch: 1.7.7 and git 68c8a83cde16160842dc6ac06f23cdadf4bbb422

### Environment information

- OS: ArchLinux
- Steam (with included Runtime)
- Steam Controller with original USB-Dongle (not BT directly)

Possibly related: https://github.com/libretro/RetroArch/issues/7990

**Comment by orbea** [comment]
Does it happen without steam?

**Comment by ml-** [comment]
Without Steam the Steam Controller behaves as a mouse and has it's usual "Desktop configuration". Meaning that the middle Button starts Big Picture, the Start-button focuses Steam (minimized RetroArch), Scrolls on the left touchpad and acts as a mouse on the right touchpad.

This makes the button mapping impossible, for example the D-Pad can't be configures because the touch of the left pad is recognized as the input.

The controller only acts as a real gamepad when the application is started through Steam itself. It might be possible to use the Steam Controller after reconfiguring the buttons using a 3rd party tool. But that changes the behavior system-wide, not for RetroArch only.

**Comment by orbea** [comment]
So its hard to tell where the issue in RetroArch ends and the potential issues in steam start...

I don't use steam or have a steam controller, would you be willing to try bisecting RetroArch to find where this might have started?

**Comment by ml-** [comment]
I don't think this ever worked properly, unless another Steam Controller user knows better. I will see whether and how I'll proceed from here. Maybe go back to the version prior vulkan and try again.

Another thing I noticed while playing around with Steam Configuration is that the "vulkan" option only works when Steam Overlay is activated for RetroArch. When it's deactivated it behaves the same way as the over video drivers. No input.

**Comment by tmtvl** [comment]
Have you tried using sc-controller? It seems to me like the controller reverts to desktop mode when RA is launched from Steam BPM. This is more likely to be an issue with Steam than with RA.

**Comment by ml-** [comment]
Hi @tmtvl,
I assume sc-controller will work without a problem. But the whole point in using Steam Big Picture Mode is to have Controller configuration handled by Steam itself for all the applications that are started from there.

For some reason RetroArch managed to bypass or ignore this completely. I haven't figured out why yet. Even more confusing is the fact that the Steam Overlay option itself has an impact on this as well.

**Comment by tmtvl** [comment]
Well, according to https://partner.steamgames.com/doc/features/overlay#requirements there are a couple of situations where the overlay won't work, namely:
* The overlay will not be active in software-rasterized games.
* The overlay is automatically disabled for 'Software' app types.

It also states that SteamAPI_Init needs to be called before initializing the OpenGL/D3D device. Maybe that's why it works fine with Vulkan. I don't know if we should look into finding a way to check if we're being launched from Steam and doing whatever needs to be done then.

**Comment by ml-** [comment]
Thank you very much @tmtvl for doing this quick research. I guess that might be the reason. At least it sounds like it.

I think the `SteamAPI_Init` call eventually will be implemented since RetroArch will come to Steam. Very curious if this will solve the Controller issue.
